src/containers/VerifyActivities.jsx

```diff
@@ -33,58 +42,128 @@ class VerifyActivities extends Component {
    */
-  static getDerivedStateFromProps(nextProps) {
-    const { societyName, societyActivities } = nextProps;
-    const activities = filterActivitiesByStatus(societyActivities, 'in review');
-    return {
-      activities,
-      societyName,
-    };
+  static getDerivedStateFromProps(props, state) {
+    if (props.userRoles.length) {
+      const { allActivities, societyName } = props;
+      const { selectedSociety } = state;
+      const userRoles = props.userRoles ? props.userRoles : [];
+      const showButtons = userRoles.length > 0 && hasAllowedRole(userRoles, [SOCIETY_SECRETARY, SUCCESS_OPS]);
+      const showMoreInfoButton = userRoles.length > 0 && hasAllowedRole(userRoles, [SUCCESS_OPS]);
+      let {
+        showTabs,
+      } = state;
+      let filteredActivities;
+      if (hasAllowedRole(userRoles, [SUCCESS_OPS])) {
+        showTabs = true;
+        filteredActivities = filterActivitiesByStatus(allActivities, PENDING)
+          .filter(activity => (activity.society.name.toLowerCase() === selectedSociety));
+      } else {
+        filteredActivities = filterActivitiesByStatus(allActivities, IN_REVIEW)
+          .filter(activity => (activity.society.name.toLowerCase() === societyName));
+      }
+
+      return {
+        filteredActivities,
+        societyName,
+        showTabs,
+        showButtons,
+        showMoreInfoButton,
+      };
+    }
+    return { ...state, userRoles: null };
   }
 
   constructor(props) {
     super(props);
     this.state = {
-      activities: [],
+      filteredActivities: [],
       showUserDetails: true,
-      societyName: '',
+      selectedStatus: PENDING,
       isSelectAllChecked: false,
       selectedActivities: [],
-      showButtons: true,
+      selectedActivity: {},
       message: null,
+      selectedSociety: 'istelle',
+      showTabs: false,
+      showMoreInfoButton: false,
+      showModal: false,
     };
   }
 
+  /**
+   * @name componentDidMount
+   * @summary Lifecycle method called when component is mounted
+   */
   componentDidMount() {
-    if (this.state.societyName) this.props.fetchSocietyInfo(this.state.societyName);
+    this.props.fetchAllActivities();
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.societyName !== this.state.societyName) {
-      this.props.fetchSocietyInfo(this.state.societyName);
+  /**
+   * @name componentDidUpdate
+   * @summary Lifecycle method called when there are updates
+   *
+   */
+  componentDidUpdate() {
+    if (!this.props.userRoles) {
+      this.props.fetchAllActivities();
     }
   }
+
+  /**
+   * handle the click event for the verify button
+   * @memberof handleClick
+   */
   handleClick = (clickAction, activityId) => {
-    this.props.verifyActivity(clickAction, activityId);
+    const { userRoles } = this.props;
+    const { APPROVE, MORE_INFO } = clickActions;
+    switch (clickAction) {
+    case APPROVE:
+      if (hasAllowedRole(userRoles, [SUCCESS_OPS])) {
```

Comment:
I suggest wrapping this case block in curly brackets since it more than one statement for readability [opinion]